model_analyzer/record/metrics_manager.py

```diff
@@ -164,44 +150,46 @@ def profile_model(self, run_config, perf_output_writer=None):
             The gpu specific and non gpu metrics
         """
 
-        collect_gpu_metrics = not run_config.model_config().cpu_only()
-        collect_cpu_metrics = not self._config.cpu_monitor_disable
```

Comment:
You may have this be set to the value of the argument `--collect-cpu-metrics`

Reply:
The logic here is changed. The `--metric` flag will be inspected for whether the CPU monitor will run or not.